### PR TITLE
Added module 03-java for angular

### PR DESCRIPTION
**Pull request by alfredya**
Added Module 03-Java for Angular.  Will work on this some more tonight, but just want to make sure I am on the right track.

### CODE REVIEW

File: handout/03-java-for-angular.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+#JavaScript for Angular
+
+Let'e begin by disecting the following example:
```
Comment (SethDavenport) [inline handout/03-java-for-angular.md]: Typo :)

File: handout/03-java-for-angular.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ 	.controller('MainCtrl', function($scope) {...});
+```
+Functions are first class citizens in JavaScript, meaning that a function can be passed into another
+function as a parameter, and a function can also be returned from a function.
+
+This concept is very important as we deal with callbacks from asynchronous function calls later in 
+this training.
+
+Also note that the scope of the controller is passed into the anonymous function. This is an example
+of dependency injection, which is one of the most powerful features on Angular.
```
Comment (SethDavenport) [inline handout/03-java-for-angular.md]: Typo: "features s/on/of/ angular"

File: handout/03-java-for-angular.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+##Closure
+
+In JavaScript, a function can reside within another function.  The function has a local (function)
```
Comment (SethDavenport) [inline handout/03-java-for-angular.md]: Is it worth talking about the fact that _only_ functions have scope in JS?

As in

while (foo) {
  // no new scope here
}

**Comment by yuri** [comment]
The "closure" section needs to provide examples that have variables define in different scopes. Also, for chained style, I like to break it up with variables to show what is going on conceptually.

File: handout/03-java-for-angular.md (hunk 3 above)
Comment (alfredya) [inline handout/03-java-for-angular.md]: Think so, will add this.

**Comment by alfredya** [comment]
Break it up with variables?  Like

var mod = angular.module('ngcourse');
mod.controller('MainCtrl', function($scope){ ... });

**Comment by yuri** [comment]
Yes, though with more meaningful names:

```js
var ngCourseModule  = angular.module('ngcourse');
ngCourseModule.controller('MainCtrl', function($scope){ ... });
```

I've found this leads to an "aha" moment for many people.

**Pull request closed without merging**